### PR TITLE
Select types returned from Graph distinctly

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

When joining tables, the same row may be selected multiple times, but some methods require exactly one row to be returned.
This PR changes the database query to select distinctly on the primary keys.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202805690238892/1203358665695497/f) _(internal)_

## 🔍 What does this change?

- Add required methods to the `OntologyPath` trait
- Replace `Compiler::with_default_selection` to pick the columns manually

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs [hunk 1]
```diff
@@ -72,7 +72,7 @@ impl From<OntologyRecord<EntityType>> for EntityTypeWithMetadata {
 impl<C: AsClient, T> Read<T> for PostgresStore<C>
 where
     T: for<'q> PersistedOntologyType<
-            Inner: PostgresQueryRecord<Path<'q>: Debug + Sync>
+            Inner: PostgresQueryRecord<Path<'q>: Debug + Send + Sync + OntologyPath>
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs:75]: Is this change required because of manually specifying the selection now?

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/statement/select.rs [hunk 2]
```diff
@@ -129,14 +129,9 @@ mod tests {
     #[test]
     fn default_selection() {
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/query/statement/select.rs:130]: This test name is not accurate anymore :)

File: packages/graph/hash_graph/tests/rest-test.http [hunk 3]
```diff
@@ -557,10 +557,10 @@ Content-Type: application/json
...
   },
   "graphResolveDepths": {
-    "dataTypeResolveDepth": 0,
-    "propertyTypeResolveDepth": 0,
-    "entityTypeResolveDepth": 0,
-    "entityResolveDepth": 0
+    "dataTypeResolveDepth": 10,
+    "propertyTypeResolveDepth": 10,
+    "entityTypeResolveDepth": 10,
+    "entityResolveDepth": 10
```
Comment (thehabbos007) [inline packages/graph/hash_graph/tests/rest-test.http:563]: Are these changes intentional?

File: packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs [hunk 4]
```diff
@@ -82,7 +82,27 @@ where
...
     async fn read<'f: 'q, 'q>(&self, filter: &'f Self::Query<'q>) -> Result<Vec<T>, QueryError> {
-        let mut compiler = SelectCompiler::with_default_selection();
+        let versioned_uri_path =
+            <<T::Inner as QueryRecord>::Path<'q> as OntologyPath>::versioned_uri();
+        let schema_path = <<T::Inner as QueryRecord>::Path<'q> as OntologyPath>::schema();
+        let owned_by_id_path = <<T::Inner as QueryRecord>::Path<'q> as OntologyPath>::owned_by_id();
+        let created_by_id_path =
+            <<T::Inner as QueryRecord>::Path<'q> as OntologyPath>::created_by_id();
+        let updated_by_id_path =
+            <<T::Inner as QueryRecord>::Path<'q> as OntologyPath>::updated_by_id();
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs:92]: The path selection is.. complicated, but also not worth investing time into fixing right now if this is in-flux :)

**Review by thehabbos007 — COMMENTED** [review]
Looks good, Tim! We already have tests for the distinctness/ordering stuff, but how do we ensure that the read trait does the right thing? Is it implicit by the HTTP tests?

File: packages/graph/hash_graph/tests/rest-test.http (hunk 3 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/tests/rest-test.http:563]: Previously, this test would have failed without this fix. Another former PR fixed this exact test but being a little bit more conservative shouldn't hurt here 🙂

Comment (thehabbos007) [reply packages/graph/hash_graph/tests/rest-test.http:563]: Cool! :)

**Review by thehabbos007 — APPROVED** [review]
